contracts/governance/test/ZeroExVotesTest.t.sol

```diff
@@ -53,6 +55,53 @@ contract ZeroExVotesTest is BaseTest {
         votes.initialize(account2, quadraticThreshold);
     }
 
+    function testShouldBeAbleToMigrate() public {
+        vm.roll(block.number + 1);
+
+        vm.startPrank(account2);
+        token.approve(address(wToken), 100e18);
+        wToken.depositFor(account2, 100e18);
+        wToken.delegate(account3);
+        vm.stopPrank();
+
+        vm.startPrank(account3);
+        token.approve(address(wToken), 200e18);
+        wToken.depositFor(account3, 200e18);
+        wToken.delegate(account3);
+        vm.stopPrank();
+
+        vm.roll(block.number + 1);
+
+        ZeroExVotesMigration newImpl = new ZeroExVotesMigration();
+        assertFalse(address(uint160(uint256(vm.load(address(votes), 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc)))) ==  address(newImpl));
+        vm.prank(account1);
+        votes.upgradeToAndCall(address(newImpl), abi.encodeWithSignature("initialize()"));
+        assertEq(address(uint160(uint256(vm.load(address(votes), 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc)))), address(newImpl));
+
+        vm.roll(block.number + 1);
+
+        vm.prank(account2);
+        wToken.transfer(address(this), 50e18);
+        vm.prank(account3);
+        wToken.transfer(address(this), 100e18);
+    }
+
+    function testShouldNotBeAbleToStopBurn() public {
+        // wrap some token
+        vm.startPrank(account2);
+        token.approve(address(wToken), 1 ether);
+        wToken.depositFor(account2, 1 ether);
+        vm.stopPrank();
+
+        // malicious upgrade
+        IZeroExVotes maliciousImpl = new ZeroExVotesMalicious();
+        votes.upgradeTo(address(maliciousImpl));
+
+        // try to withdraw withdraw
+        vm.prank(account2);
+        wToken.withdrawTo(account2, 1 ether);
+    }
```

Comment:
If `ZRXWrappedToken` were to use `functionCallWithGas` or similar, this scenario could be avoided.

Reply:
Just to be clear this scenario is only possible if the voting implementation is upgraded to a malicious one through governance?

Reply:
Correct